application/restore.sh

```diff
@@ -1,45 +1,147 @@
-#!/usr/bin/env sh
+#!/usr/bin/env bash
```

Comment:
Requiring bash (and not just any shell) is potentially a backward-compatibility breaking change. Can it be done without restricting the script to bash?